js/client-battle-tooltips.js

```diff
@@ -409,6 +418,260 @@ var BattleTooltips = (function () {
+					modifier = goodBoostTable[pokemon.boosts[statName]];
+				} else {
+					var badBoostTable = [1, 0.67, 0.5, 0.4, 0.33, 0.29, 0.25];
+					// var badBoostTable = ['Normal', '&minus;1', '&minus;2', '&minus;3', '&minus;4', '&minus;5', '&minus;6'];
+					modifier =  badBoostTable[-pokemon.boosts[statName]];
+				}
+
+			}
+		}
+
+		// check for burn, paralysis, guts, quick feet
+		if (pokemon.status) {
+			if (statName === 'atk') {
+				if (this.battle.gen > 2 && pokemon.ability === 'Guts')  {
+					modifier = modifier * 1.5;
+				} else if (pokemon.status === 'brn') {
+					modifier = modifier * 0.5;
+				}
+			}
+
+			if (statName === 'spe') {
+				if (this.battle.gen > 2 && pokemon.ability === 'Quick Feet') {
+					modifier = modifier * 1.5;
+				} else if (pokemon.status === 'par') {
+					modifier = modifier * 0.25;
+				}
+			}
+		}
+
+		// gen 1 stuff doesn’t work yet
+		if (this.battle.gen > 1) {
+			var item = '';
+			if (myPokemon.item) {
+				item = Tools.getItem(myPokemon.item).name;
+			}
+
+
+			// check for light ball, thick club, metal/quick powder
+			// the only stat modifying items in gen 2 were light ball, thick club, metal powder. gen 1 had none
+			if (item === 'Light Ball') {
+				if (pokemon.baseSpecies === 'Pikachu' && (statName === 'atk' || statName === 'spa')) {
+					modifier = modifier * 2;
+				}
+			}
+
+			if (item === 'Thick Club') {
+				if ((pokemon.baseSpecies === 'Marowak' || pokemon.baseSpecies === 'Cubone') && statName === 'atk') {
+					modifier = modifier * 2;
+				}
+			}
+
+			if (pokemon.baseSpecies === 'Ditto') {
+				if (!('transform' in pokemon.volatiles)) {
+					if (item === 'Quick Powder' && statName === 'spe') {
+						modifier = modifier * 2;
+					}
+					if (item === 'Metal Powder' && (statName === 'spd' || statName === 'def')) {
+						modifier = modifier * 1.5;
+					}
+
+				}
+			}
+
+			// check abilities other than Guts and Quick Feet
+			// check items other than light ball, thick club, metal/quick powder
+			if (this.battle.gen > 2) {
+				var ability = Tools.getAbility(myPokemon.baseAbility).name;
+				if (this.battle.weather) {
+					// Check if you have an anti weather ability
+					var noWeatherAbility = !!(ability in {'Air Lock': 1, 'Cloud Nine': 1});
+					// If you don't, check if the opponent has it afterwards.
+					if (!noWeatherAbility) {
+						for (var i = 0; i < this.battle.yourSide.active.length; i++) {
+							if (this.battle.yourSide.active[i] && this.battle.yourSide.active[i].ability in {'Air Lock': 1, 'Cloud Nine': 1}) {
+								noWeatherAbility = true;
+								break;
+							}
+						}
+					}
+				}
+
+				if (item === 'Choice Band' && statName === 'atk') {
+					modifier = modifier * 1.5;
+				}
+				if (ability === 'Toxic Boost' && (pokemon.status === 'tox' || pokemon.status === 'psn') && statName === 'atk') {
+					modifier = modifier * 1.5;
+				}
+				if (ability === 'Pure Power' || pokemon.ability === 'Huge Power' && statName === 'atk') {
```

Comment:
Should be

`(ability === 'Pure Power' || pokemon.ability === 'Huge Power') && statName === 'atk'`

The way it's currently implemented, Pure Power doubles all stats.